build/azDevOps/azure/ensono/release-notes/publish-release-notes.yml

```diff
@@ -1,55 +1,55 @@
 parameters:
-  - name: version_major
-    type: string
-    default: '1'
-  - name: version_minor
-    type: string
-    default: '0'
+  - name: enable
+    type: boolean
 
-jobs:
-  - job: GenerateAndPublish
-    uses:
-      repositories: [wiki] # work around for security issue for pipelines to access the WIKI repo
-    steps:
-      - task: XplatGenerateReleaseNotes@4
-        displayName: Generate Release Notes
-        inputs:
-          outputfile: "$(Build.ArtifactStagingDirectory)/releasenotes.md"
-          templateLocation: "File"
-          templatefile: "$(System.DefaultWorkingDirectory)/build/azDevOps/azure/ensono/release-notes/handlebar/releaseNoteTemplate.md"
-          customHandlebarsExtensionFile: "custom-extensions.js"
-          customHandlebarsExtensionFolder: "$(System.DefaultWorkingDirectory)/build/azDevOps/azure/ensono/release-notes/handlebar/extensions"
-          checkStage: false # Looks for the last successful release.  Else release notes will be compared against the previous release.
-          replaceFile: false
-          getParentsAndChildren: false
-          getAllParents: false
-          getIndirectPullRequests: false
-          stopOnError: true
-          considerPartiallySuccessfulReleases: true
-          checkForManuallyLinkedWI: true # Picks up items manually linked rather than via commit message
-          wiqlFromTarget: "WorkItems"
-          dumpPayloadToFile: true
-          dumpPayloadFileName: $(Build.ArtifactStagingDirectory)/release-notes-payload.json
-      - task: PublishBuildArtifacts@1
-        displayName: "Publish Release Notes Payload"
-        inputs:
-          pathtoPublish: "$(Build.ArtifactStagingDirectory)/release-notes-payload.json"
-          artifactName: "release-notes-payload.json"
+stages:
+  - ${{ if eq(parameters.enable, true) }}:
+      - stage: ReleaseNotes
+        dependsOn:
+          - Dev
+          - Prod
+        condition: or(succeeded('Dev'), succeeded('Prod'))
```

Comment:
May best to parameterise these. All our pipeline use `Dev` and `Prod` but it's likely more stages will be added. Having these as parameters would mean only the main pipeline file needs changes when added new stages.

Reply:
Parameterised